### PR TITLE
don't use error message but check returncode of the command

### PR DESCRIPTION
**Pull request by Segaja**
using return code to find out if a folder is under git version control.

### CODE REVIEW

**Comment by Felixoid** [comment]
Hello @olivierverdier 
Could you please merge this to fix annoying prompt in each folder on the shell?

File: gitstatus.py [hunk 1]
```diff
@@ -10,9 +10,7 @@
 gitsym = Popen(['git', 'symbolic-ref', 'HEAD'], stdout=PIPE, stderr=PIPE)
 branch, error = gitsym.communicate()
 
-error_string = error.decode('utf-8')
-
-if 'fatal: Not a git repository' in error_string:
+if 0 != gitsym.returncode:
```
Comment (Felixoid) [inline gitstatus.py:13]: this change broke prompt like `(:de68074|✔)` on detached HEAD

File: gitstatus.py [hunk 2]
```diff
@@ -10,9 +10,7 @@
 gitsym = Popen(['git', 'symbolic-ref', 'HEAD'], stdout=PIPE, stderr=PIPE)
```
Comment (Felixoid) [inline gitstatus.py:10]: There is most quick way to get info are we in git repo or not
```
isgit = Popen(['git', 'name-rev'], stdout=PIPE, stderr=PIPE)
_, _ = isgit.communicate()                                  
if isgit.returncode != 0:                                   
    sys.exit(0)                                             
```
IMO it should be right after `import sys`

**Review by Felixoid — CHANGES_REQUESTED** [review]

File: gitstatus.py (hunk 1 above)
Comment (Felixoid) [reply gitstatus.py:13]: If you'll add `if isgit.returncode != 0:`, then this code could be deleted

**Comment by Segaja** [comment]
The problem is that the `git symbolic-ref HEAD` command also found the branch name and that is used later. so either I have to add another command to run or figure out how to do this with one command without breaking detached heads.

**Pull request closed without merging**